Extension/src/LanguageServer/configurations.ts

```diff
@@ -989,15 +989,12 @@ export class CppProperties {
                 // However, if compileCommands are used and compilerPath is explicitly set, it's still necessary to resolve variables in it.
                 if (configuration.compilerPath === "${default}") {
                     configuration.compilerPath = settings.defaultCompilerPath ?? undefined;
+                    configuration.compilerPathIsExplicit = settings.defaultCompilerPath === null && configuration.compilerPath === undefined;
                 }
-                if (configuration.compilerPath === null) {
-                    configuration.compilerPath = undefined;
-                    configuration.compilerPathIsExplicit = true;
-                } else if (configuration.compilerPath !== undefined) {
+
+                if (configuration.compilerPath !== undefined) {
                     configuration.compilerPath = util.resolveVariables(configuration.compilerPath, env);
                     configuration.compilerPathIsExplicit = true;
-                } else {
```

Comment:
I think this `else` is still required. Maybe it should be

```
} else if (settings.defaultCompilerPath === null) {
    configuration.compilerPathIsExplicit = true;
} else {
    configuration.compilerPathIsExplicit = false;
}
```

with `configuration.compilerPathIsExplicit = settings.defaultCompilerPath === null && configuration.compilerPath === undefined;` removed.

Reply:
Oops, sorry for telling you to do the wrong thing, but I realized my suggestion was wrong.

There's a problem because `settings.defaultCompilerPath === null` should only be checked if `configuration.compilerPath === "${default}"` is true, so maybe just add a check for that.

Reply:
You may need to add a bool that is set to true in `if (configuration.compilerPath === "${default}") {` and check that bool.

Reply:
You should ask @Colengms what the intent was. PR here: https://github.com/microsoft/vscode-cpptools/pull/5855/files

Reply:
@bobbrow Yeah, I think that's what the plan was before checking it in. From what I can tell, the code as originally written by Colen never executed the code in the `if` block.

Reply:
I'm confused.  `null` is a valid value that `compilerPath` can have in a configuration.  Wouldn't the correct change here be to keep the `compilerPath === null` check (maybe remove where it gets normalized to undefined), so we are properly supporting users who set the `compilerPath` to `null` in their `c_cpp_properties.json`?  That might also require ensuring other reference to the variable properly handle null, where it was not explicitly part of the type before. (This it probably wound up being loaded as null previously, despite the type not being correct)

Reply:
@Colengms I think you're approach is right, we were just unsure about the propagation of null values initially.